### ISSUE TITLE
Handling Boundary/Ghost Cells

### ISSUE THREAD

**Issue by sjordan29** [EcohydrologyTeam/ClearWater-riverine#1]
@aufdenkampe, @jrutyna - some questions about boundary conditions to keep in mind as you review the code. 

**Ghost Cells Receiving Flow**
In RAS2D output, all ghost cells (boundary cells) have a volume of 0. However, this results in an error in the sparse matrix solver because nothing can leave the mesh, even when it should (e.g. in our [box model](https://github.com/EnvironmentalSystems/ClearWater-riverine/tree/main/tests/input_files/Box-Model).) Therefore, we must calculate the 'volume' in the ghost cells as the total flow across the edge between the real cell and ghost cell in a given timestep: [implemented here](https://github.com/EnvironmentalSystems/ClearWater-riverine/blob/main/examples/ras2dwq.py#L252-L280). These ghost volumes are then inserted into the sparse LHS matrix on the diagonals ([implemented here](https://github.com/EnvironmentalSystems/ClearWater-riverine/blob/main/examples/ras2dwq.py#L551-L558)), so that material can move out of the grid when there is flow across an edge to a ghost cell. 

**Ghost Cells Flowing Into the Mesh**
In order to set boundary conditions of a concentration of a certain contaminant/ghost cell, we also need to consider the volume of that cell when solving our sparse matrix. I calculate the volume of these ghost cells in a similar way as above, where I calculate teh flow across the edge between the real cell and the ghost cell in a given timestep, but now I assign that volume to the prior timestep for flow coming into the model mesh ([implemented here](https://github.com/EnvironmentalSystems/ClearWater-riverine/blob/main/examples/ras2dwq.py#L286-L310)).

**Putting this information in the Sparse Matrix**
I know I need to consider _ghost cells flowing into the mesh_ on the right hand side of the sparse matrix problem in order to have any boundary condition concentrations flow into the mesh. But how should I handle the _ghost cells receiving flow_ in this right hand side? Should the concentration reset to 0 at every timestep? Should users be defining ambient boundary conditions for every timestep? Currently, I include both of these ghost cell volumes in the [RHS class] (https://github.com/EnvironmentalSystems/ClearWater-riverine/blob/main/examples/ras2dwq.py#L648-L649)

**TLDR**
* How should boundary conditions be handled for ghost cells?
* Is there any difference in how boundary conditions should be handled for ghost cells that are flowing into the mesh vs receiving flow from the mesh?

**Comment by sjordan29** [comment]
Addressed in Release 0.4.0
